src/memcached/cache_impl_test.go

```diff
@@ -0,0 +1,93 @@
+package memcached
+
+import (
+	"errors"
+	"github.com/bradfitz/gomemcache/memcache"
+	"github.com/stretchr/testify/assert"
+	"net"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+
+	mock_srv "github.com/envoyproxy/ratelimit/test/mocks/srv"
+)
+
+func TestRefreshServersSetsServersOnEmptyServerList(t *testing.T) {
+	assert := assert.New(t)
+
+	mockSrv := "_memcache._tcp.example.org"
+	mockMemcacheHostPort := "127.0.0.1:11211"
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockSrvResolver := mock_srv.NewMockSrvResolver(ctrl)
+
+	mockSrvResolver.EXPECT().ServerStringsFromSrv(gomock.Eq(mockSrv)).Return([]string{mockMemcacheHostPort}, nil)
+
+	serverList := new(memcache.ServerList)
+
+	refreshServers(serverList, mockSrv, mockSrvResolver)
+
+	actualMemcacheHosts := []string{}
+
+	serverList.Each(func(addr net.Addr) error {
+		actualMemcacheHosts = append(actualMemcacheHosts, addr.String())
+		return nil
+	})
+
+	assert.Equal([]string{mockMemcacheHostPort}, actualMemcacheHosts)
+}
+
+func TestRefreshServerSetsServersOnNonEmptyServerList(t *testing.T) {
```

Comment:
nit: i think more descriptive name this test case would be `...OverridesServersOnNonEmptyServerList`